lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -411,53 +398,85 @@ def calculate_wet_bulb_temperature(self, temperature, relative_humidity,
                 Cube of air pressures (Pa).
 
         Returns:
-            wbt (iris.cube.Cube):
+            iris.cube.Cube:
                 Cube of wet bulb temperature (K).
 
         """
-        precision = np.full(temperature.data.shape, self.precision)
-
-        # Set units of input diagnostics.
         relative_humidity.convert_units(1)
         pressure.convert_units('Pa')
         temperature.convert_units('K')
 
+        precision = np.full(temperature.data.shape, self.precision)
+        wbt = temperature.copy()
+        wbt.rename('wet_bulb_temperature')
+
+        latent_heat = Utilities.latent_heat_of_condensation(temperature)
+
+        wbt_data = self._calculate_wbt(
+            latent_heat.data, precision, pressure.data,
+            relative_humidity.data, temperature.data)
+
+        wbt.data = wbt_data
+        return wbt
+
+    def _calculate_wbt(self, latent_heat, precision, pressure,
+                       relative_humidity, temperature):
+        """Calculates the wet bulb temperature. A Newton iterator is
+        used to minimise the gradient of enthalpy against temperature.
+
+        Args:
+            latent_heat (numpy.ndarray):
+                Array of temperature adjusted latent heat of
+                condensation (J kg-1).
+            precision (numpy.ndarray):
+                The precision to which the Newton iterator must converge before
+                returning wet bulb temperatures. In the shape of temperature.
+            pressure (numpy.ndarray):
+                Array of air Pressure (Pa).
+            relative_humidity (numpy.ndarray):
+                Array of relative humidities (1).
+            temperature (numpy.ndarray):
+                Array of air temperature (K).
+
+        Returns:
```

Comment:
If this is going to return wbt_data it should probably say:
Returns:
    wbt_data (numpy.nadarray)